<h1 style="text-align:center; color:#2e86de;">🚀 TRAINING 7</h1>

# ***1. Chargement des données***

In [22]:
import pandas as pd
import numpy as np  


In [23]:
students_social_media_addiction = pd.read_csv("students_social_media_addiction.csv")
students_social_media_addiction.head()

Student_ID  Age  Gender Academic_Level     Country  Avg_Daily_Usage_Hours  \
0           1   19  Female  Undergraduate  Bangladesh                    5.2   
1           2   22    Male       Graduate       India                    2.1   
2           3   20  Female  Undergraduate         USA                    6.0   
3           4   18    Male    High School          UK                    3.0   
4           5   21    Male       Graduate      Canada                    4.5   

  Most_Used_Platform Affects_Academic_Performance  Sleep_Hours_Per_Night  \
0          Instagram                          Yes                    6.5   
1            Twitter                           No                    7.5   
2             TikTok                          Yes                    5.0   
3            YouTube                           No                    7.0   
4           Facebook                          Yes                    6.0   

   Mental_Health_Score Relationship_Status  Conflicts_Over_Social_Media  \
0                    6     In Relationship                            3   
1                    8              Single                            0   
2                    5         Complicated                            4   
3                    7              Single                            1   
4                    6     In Relationship                            2   

   Addicted_Score  
0               8  
1               3  
2               9  
3               4  
4               7

In [24]:
students_social_media_addiction.describe()

Student_ID         Age  Avg_Daily_Usage_Hours  Sleep_Hours_Per_Night  \
count  705.000000  705.000000             705.000000             705.000000   
mean   353.000000   20.659574               4.918723               6.868936   
std    203.660256    1.399217               1.257395               1.126848   
min      1.000000   18.000000               1.500000               3.800000   
25%    177.000000   19.000000               4.100000               6.000000   
50%    353.000000   21.000000               4.800000               6.900000   
75%    529.000000   22.000000               5.800000               7.700000   
max    705.000000   24.000000               8.500000               9.600000   

       Mental_Health_Score  Conflicts_Over_Social_Media  Addicted_Score  
count           705.000000                   705.000000      705.000000  
mean              6.226950                     2.849645        6.436879  
std               1.105055                     0.957968        1.587165  
min               4.000000                     0.000000        2.000000  
25%               5.000000                     2.000000        5.000000  
50%               6.000000                     3.000000        7.000000  
75%               7.000000                     4.000000        8.000000  
max               9.000000                     5.000000        9.000000

# ***2. Déscription complète de ces données***

### **2.A - Les variables sont** : 
- Student_ID
- Age
- Avg_Daily_Usage_Hours
- Sleep_Hours_Per_Night 
- Mental_Health_Score    
- Conflicts_Over_Social_Media
- Addicted_Score

### **2.B -** Il y a pas de variable ayant **une donnée manquante** 

### **2.C -** Toutes les variables sont de type **Integer**

 ### **2.D - Les valeurs minimales et les valeurs maximales de chaque variable sont présentées dans le tableau ci-dessous :**<br/>
| Variable                      | Valeur minimale | Valeur maximale |
|-------------------------------|-----------------|-----------------|
| Student_ID                    | 1               | 705             |
| Age                           | 18              | 24              |
| Avg_Daily_Usage_Hours         | 1.5             | 8.5             |
| Sleep_Hours_Per_Night         | 3.8             | 9.6             |
| Mental_Health_Score           | 4               | 9               |
| Conflicts_Over_Social_Media   | 0               | 5               |
| Addicted_Score                | 2               | 5               |

# ***3. Extraction de quelques données***

### **3.A - Extraction des âges**

In [25]:
ages_des_eleves = students_social_media_addiction["Age"]
print(ages_des_eleves)

0      19
1      22
2      20
3      18
4      21
       ..
700    20
701    23
702    21
703    24
704    19
Name: Age, Length: 705, dtype: int64


### **3.B - Extraction des niveaux d'addiction**

In [26]:
niveaux_de_addiction = students_social_media_addiction["Addicted_Score"]
print(niveaux_de_addiction)

0      8
1      3
2      9
3      4
4      7
      ..
700    5
701    9
702    7
703    4
704    8
Name: Addicted_Score, Length: 705, dtype: int64


# ***4. Statistiques***

### **$4.A_1$ - La plateforme la plus utilisée par tous les élèves**

In [27]:
# Trouver la plateforme la plus utilisée par tous les élèves
plateforme_plus_utilisee_general = students_social_media_addiction["Most_Used_Platform"].value_counts().idxmax()
print("La plateforme la plus utilisée par les élèves est :", plateforme_plus_utilisee_general)

La plateforme la plus utilisée par les élèves est : Instagram


#### **Confirmation de la plateforme la plus utilisée par tous les élèves**

In [28]:
# Calculer le nombre d'élèves utilisant chaque plateforme
plateforme_counts = students_social_media_addiction["Most_Used_Platform"].value_counts().reset_index()
plateforme_counts.columns = ["Plateformes", "Nombre d'élèves qui l'utilisent"]
print(plateforme_counts)

   Plateformes  Nombre d'élèves qui l'utilisent
0    Instagram                              249
1       TikTok                              154
2     Facebook                              123
3     WhatsApp                               54
4      Twitter                               30
5     LinkedIn                               21
6       WeChat                               15
7     Snapchat                               13
8         LINE                               12
9    KakaoTalk                               12
10   VKontakte                               12
11     YouTube                               10


### **$4.A_2$ - Les plateformes ayant un effet sur les résultats académiques des élèves**

In [29]:
# On filtre les élèves dont les résultats sont affectés par les réseaux sociaux
plateformes_avec_effet = students_social_media_addiction[
    students_social_media_addiction["Affects_Academic_Performance"] == "Yes"
]

# On regroupe par plateforme pour compter le nombre d'élèves impactés
plateformes_avec_effet = (
    plateformes_avec_effet["Most_Used_Platform"]
    .value_counts()
    .reset_index()
)

# On renomme les colonnes pour plus de clarté
plateformes_avec_effet.columns = ["Plateforme", "Nombre d'élèves impactés"]

# Affichage du tableau
print(plateformes_avec_effet)

  Plateforme  Nombre d'élèves impactés
0  Instagram                       172
1     TikTok                       144
2   WhatsApp                        54
3   Facebook                        37
4   Snapchat                        12
5  KakaoTalk                        12
6    Twitter                        11
7     WeChat                         8
8    YouTube                         3


### **$4.B_1$ - La plateforme la plus utilisée par les tous élèves en Inde**

In [30]:
eleves_en_inde = students_social_media_addiction[students_social_media_addiction["Country"] == "India"]
plateforme_plus_utilisee = eleves_en_inde["Most_Used_Platform"].value_counts().idxmax()
print("La plateforme la plus utilisée par tous les élèves en Inde est :", plateforme_plus_utilisee)

La plateforme la plus utilisée par tous les élèves en Inde est : WhatsApp


### **Confirmation de la plateforme la plus utilisée par tous les élèves en Inde**

In [31]:
# Compter le nombre d'élèves pour chaque plateforme en Inde
plateforme_counts = eleves_en_inde["Most_Used_Platform"].value_counts().reset_index()
plateforme_counts.columns = ["Plateformes utilisés en inde", "Nombre d'élèves qui l'utilisent"]

print(plateforme_counts)

  Plateformes utilisés en inde  Nombre d'élèves qui l'utilisent
0                     WhatsApp                               25
1                     Facebook                               13
2                    Instagram                                9
3                       TikTok                                5
4                      Twitter                                1


### **$4.B_2$ - Les plateformes ayant un effet sur les résultats académiques des élèves en Inde**

In [32]:
# Filtrer les élèves qui vivent en Inde
eleves_en_inde = students_social_media_addiction[
    students_social_media_addiction["Country"] == "India"
]

# Parmi eux, filtrer ceux dont les résultats académiques sont affectés
plateformes_effet_inde = eleves_en_inde[
    eleves_en_inde["Affects_Academic_Performance"] == "Yes"
]["Most_Used_Platform"].value_counts().reset_index()

# Renommer les colonnes pour plus de clarté
plateformes_effet_inde.columns = ["Plateforme", "Nombre d'élèves impactés en Inde"]

# Afficher le tableau final
print(plateformes_effet_inde)

  Plateforme  Nombre d'élèves impactés en Inde
0   WhatsApp                                25
1  Instagram                                 9
2     TikTok                                 5
3   Facebook                                 3


### **$4.C_{1}$ - La plateforme la plus utilisée par les tous élèves au USA**

In [33]:
eleves_usa = students_social_media_addiction[students_social_media_addiction["Country"] == "USA"]
plateforme_plus_utilisee_usa = eleves_usa["Most_Used_Platform"].value_counts().idxmax()
print("La plateforme la plus utilisée par tous les élèves aux USA est :", plateforme_plus_utilisee_usa)

La plateforme la plus utilisée par tous les élèves aux USA est : Instagram


### **Confirmation de la plateforme la plus utilisée par tous les élèves au USA**

In [34]:
# Compter le nombre d'élèves pour chaque plateforme aux USA
plateforme_counts_usa = eleves_usa["Most_Used_Platform"].value_counts().reset_index()
plateforme_counts_usa.columns = ["Plateformes utilisés aux USA", "Nombre d'élèves qui l'utilisent"]
print(plateforme_counts_usa)

  Plateformes utilisés aux USA  Nombre d'élèves qui l'utilisent
0                    Instagram                               19
1                       TikTok                               16
2                      Twitter                                5


### **$4.C_2$ - Les plateformes ayant un effet sur les résultats académiques des élèves au USA**

In [36]:
eleves_usa = students_social_media_addiction[students_social_media_addiction["Country"] == "USA"]
plateformes_effet_usa = eleves_usa[eleves_usa["Affects_Academic_Performance"] == "Yes"]["Most_Used_Platform"].value_counts().reset_index()
plateformes_effet_usa.columns = ["Plateforme", "Nombre d'élèves impactés aux USA"]
print(plateformes_effet_usa)

  Plateforme  Nombre d'élèves impactés aux USA
0  Instagram                                19
1     TikTok                                16
2    Twitter                                 5


## ***5. Comparaison de l'addition des filles à celle des garçons***

In [39]:
# Comparer l'addiction des filles et des garçons
addiction_par_genre = students_social_media_addiction.groupby("Gender")["Addicted_Score"].mean()
print("Score moyen d'addiction par genre :")
print(round(addiction_par_genre, 2))

Score moyen d'addiction par genre :
Gender
Female    6.52
Male      6.36
Name: Addicted_Score, dtype: float64


# ***6. L'âge pour lequel l'addiction est plus forte***

In [ ]:
addiction_par_age = students_social_media_addiction.groupby("Age")["Addicted_Score"].mean()
age_max_addiction = addiction_par_age.idxmax()
score_max_addiction = addiction_par_age.max()
print(f"L'âge avec le score d'addiction le plus élevé est {age_max_addiction} ans")

L'âge avec le score d'addiction le plus élevé est 18 ans
